source/tilecameragenerator.ts

```diff
@@ -0,0 +1,491 @@
+                this._iterationAlgorithmIndices.push([x, y]);
+                numberOfFoundIndices++;
+            }
+        }
+    }
+
+    /**
+     * Fills the iterationAlgorithmIndices with the table indices
+     * from the HilbertCurve-Iteration-Algorithm.
+     */
+    protected fillIterationIndicesWithHilbert(): void {
+        this._iterationAlgorithmIndices = [];
+        const tableSize = this.numberOfXTiles() > this.numberOfYTiles() ? this.numberOfXTiles() : this.numberOfYTiles();
+        const recursionDepth = Math.ceil(Math.log2(tableSize));
+        const tableSizeNextPowerOfTwo = Math.pow(2, recursionDepth);
+        this.genHilbertIndices(0, 0, tableSizeNextPowerOfTwo, 0, 0, tableSizeNextPowerOfTwo, recursionDepth);
+    }
+
+    /**
+     * Recursively fills the iterationAlgorithmIndices with the HilbertCurve.
+     * Do not use this method. Use fillIterationIndicesWithHilbert() instead.
+     */
+    protected genHilbertIndices(x: number, y: number,
+        xi: number, xj: number, yi: number, yj: number, depth: number): void {
+        if (depth <= 0) {
+            x = x + (xi + yi - 1) / 2;
+            y = y + (xj + yj - 1) / 2;
+            if (x < this.numberOfXTiles() && y < this.numberOfYTiles()) {
+                this._iterationAlgorithmIndices.push([x, y]);
+            }
+        } else {
+            this.genHilbertIndices(x, y, yi / 2, yj / 2, xi / 2, xj / 2, depth - 1);
+            this.genHilbertIndices(x + xi / 2, y + xj / 2, xi / 2, xj / 2, yi / 2, yj / 2, depth - 1);
+            this.genHilbertIndices(x + xi / 2 + yi / 2, y + xj / 2 + yj / 2, xi / 2, xj / 2, yi / 2, yj / 2, depth - 1);
+            this.genHilbertIndices(x + xi / 2 + yi, y + xj / 2 + yj, -yi / 2, -yj / 2, -xi / 2, -xj / 2, depth - 1);
+        }
+    }
+
+    /**
+     * Returns the padded tileSize.
+     * @returns - The padded tileSize.
+     */
+    protected getPaddedTileSize(): [number, number] {
+        return [this.padding[1] + this.padding[3] + this.tileSize[0],
+        this.padding[0] + this.padding[2] + this.tileSize[1]];
+    }
+
+    /**
+     * Used for Iterator behavior. It sets the tile to 0 (first tile) if the value of tile is negative
+     * and therefore initiates the iteration.
+     * Otherwise it increments the tile and calls update to directly change the camera.
+     * If the tile index would get out of range it returns false to indicate, that all tiles were rendered.
+     * @returns - If the camera has been set to a next tile.
+     */
+    public nextTile(): boolean {
+        if (this.tile >= this.numberOfTiles() - 1) {
+            return false;
+        }
+        if (this.tile < 0) {
+            this.tile = -1;
+        }
+        ++this.tile;
+        this.update();
+        return true;
+    }
+
+    /**
+     * Returns if tiles still need to be rendered.
+     */
+    public hasNextTile(): boolean {
+        if (this.tile < this.numberOfTiles() - 1 && this.tile >= 0) {
```

Comment:
Please use the more concise form `return this.tile < this.numberOfTiles() - 1 && this.tile >= 0;`